lib/core/config.rb

```diff
@@ -1,32 +1,59 @@
 # frozen_string_literal: true
 
+require_relative "helpers"
+
 class Config # rubocop:disable Metrics/ClassLength
-  attr_reader :config, :current,
-              :org, :org_comes_from_env, :app, :apps, :app_dir,
-              # command line options
-              :args, :options
+  attr_reader :args, :options, :app_dir,
+              :apps, :org, :org_comes_from_env, :app, :app_comes_from_env
+
+  include Helpers
 
-  CONFIG_FILE_LOCATIION = ".controlplane/controlplane.yml"
+  CONFIG_FILE_LOCATION = ".controlplane/controlplane.yml"
 
-  def initialize(args, options)
+  def initialize(args, options, required_options)
     @args = args
     @options = options
-    @org = options[:org]&.strip
-    @org_comes_from_env = true if ENV.fetch("CPLN_ORG", nil)
-    @app = options[:app]&.strip
+    @required_options = required_options
 
-    load_app_config
+    # The order here is important
+    load_config_file
     load_apps
+    load_app
+    load_org
+
+    ensure_required_options!
+
+    warn_deprecated_config_keys
 
     Shell.verbose_mode(options[:verbose])
   end
 
-  def [](key)
+  def global(key = nil)
+    key ? @global_config&.dig(key) : @global_config
+  end
+
+  def global!(key = nil)
+    ensure_global_config!
+
+    return @global_config unless key
+
+    raise "Can't find config key '#{key}' in 'controlplane.yml'." unless @global_config.key?(key)
+
+    @global_config[key]
+  end
+
+  def current(key = nil)
+    key ? @current_config&.dig(key) : @current_config
```

Comment:
1. why not just make this lazy and always take a param of the key
2. then current_config is a regular method that will throw if the config can't be parsed. 
3. Given `current_config` can't do anything if the config doesn't exist, we probably don't need to name as `current_config!`

I hate methods that have 2 types of return values based on the argument!

@dzirtusss agree?

Reply:
Let's handle this in a separate PR.